t/plugin/proxy-mirror.t

```diff
@@ -445,3 +455,163 @@ GET /t
+                   }]]
+                   )
+
+               if code >= 300 then
+                   ngx.status = code
+               end
+               ngx.say(body)
+           }
+       }
+--- request
+GET /t
+--- error_code: 200
+--- response_body
+passed
+--- no_error_log
+[error]
+
+
+
+=== TEST 15: hit route with sample_ratio 1
+--- request
+GET /hello?sample_ratio=1
+--- error_code: 200
+--- response_body
+hello world
+--- error_log_like eval
+qr/uri: \/hello\?sample_ratio=1/
+
+
+
+=== TEST 16: set mirror requests sample_ratio to 0.5
+--- config
+       location /t {
+           content_by_lua_block {
+                local http = require("resty.http")
+               local core = require("apisix.core")
+               local t = require("lib.test_admin").test
+               local code, body = t('/apisix/admin/routes/1',
+                    ngx.HTTP_PUT,
+                    [[{
+                        "plugins": {
+                            "proxy-mirror": {
+                               "host": "http://127.0.0.1:1986",
+                               "sample_ratio": 0.5
+                            }
+                        },
+                        "upstream": {
+                            "nodes": {
+                                "127.0.0.1:1980": 1
+                            },
+                            "type": "roundrobin"
+                        },
+                        "uri": "/hello"
+                   }]]
+                   )
+
+               if code == 200 then
+                    -- reset count
+                    local count = 0
+                    local httpc = http.new()
+                    local url = "http://127.0.0.1:1980/stat_count?action=reset"
+                    local res, err = httpc:request_uri(url, {method = "GET"})
+                    if not res then
+                        core.log.error(err)
+                    else
+                        core.log.info("reset the mirror request stat count to " .. res.body)
+                        count = res.body
+                    end
+
+                   for i = 1, 200 do
+                       t('/hello?sample_ratio=0.5', ngx.HTTP_GET)
+                   end
+
+                    url = "http://127.0.0.1:1980/stat_count"
+                    res, err = httpc:request_uri(url, {method = "GET"})
+                    if not res then
+                        core.log.error(err)
+                    else
+                        core.log.info("the mirror request stat count is " .. res.body)
+                        count = res.body
+                    end
+                   assert(count >= 75 and count <= 125)
+               elseif code >= 300 then
+                   ngx.status = code
+               end
+               ngx.say(body)
+           }
+       }
+--- request
+GET /t
+--- error_code: 200
+--- response_body
+passed
+--- error_log_like eval
+qr/(uri: \/hello\?sample_ratio=0\.5){75,125}/
+--- timeout: 60
```

Comment:
We can send requests parallelly like https://github.com/apache/apisix/blob/master/docs/en/latest/internal/testing-framework.md#send-request

Remember to use a high worker_connections in this case.

Reply:
okay, the test case has changed to send requests parallelly, but the ci log don't show more info about the test status 500, could you have a look at it~ @spacewander